project/MimaExcludes.scala

```diff
@@ -145,6 +145,15 @@ object MimaExcludes {
         // SPARK-12510 Refactor ActorReceiver to support Java
         ProblemFilters.exclude[AbstractClassProblem]("org.apache.spark.streaming.receiver.ActorReceiver")
       ) ++ Seq(
+        // SPARK-12895 Implement TaskMetrics using accumulators
+        ProblemFilters.exclude[MissingMethodProblem]("org.apache.spark.TaskContext.internalMetricsToAccumulators"),
+        ProblemFilters.exclude[MissingMethodProblem]("org.apache.spark.TaskContext.collectInternalAccumulators"),
+        ProblemFilters.exclude[MissingMethodProblem]("org.apache.spark.TaskContext.collectAccumulators")
+      ) ++ Seq(
+        // SPARK-12896 Send only accumulator updates to driver, not TaskMetrics
+        ProblemFilters.exclude[IncompatibleMethTypeProblem]("org.apache.spark.Accumulable.this"),
```

Comment:
One annoying limitation of MiMa is that I think this will exclude _any_ update to the constructors that we make during the 2.x cycle, so we'll have to be careful to not accidentally break binary compatibility for the other constructors.

Note to self in review: make sure that we preserved compatibility for old public constructors here.

@andrewor14, was this exclude because the `private` constructor wasn't appropriately ignored by MiMa?